src/main/java/rx/internal/operators/OperatorObserveOn.java

```diff
@@ -148,31 +156,39 @@ public void onError(final Throwable e) {
 
         protected void schedule() {
             if (COUNTER_UPDATER.getAndIncrement(this) == 0) {
-                recursiveScheduler.schedule(new Action0() {
-
-                    @Override
-                    public void call() {
-                        pollQueue();
-                    }
-
-                });
+                Action0 a = pollQueueAction;
+                if (a == null) {
+                    a = new PollQueueAction();
+                    pollQueueAction = a;
+                }
+                recursiveScheduler.schedule(a);
             }
         }
 
         // only execute this from schedule()
         private void pollQueue() {
             int emitted = 0;
+
+            final ScheduledUnsubscribe u = scheduledUnsubscribe;
+            final RxRingBuffer q = queue;
+            final Subscriber<? super T> child = this.child;
+            final NotificationLite<T> on = this.on;
+            @SuppressWarnings("rawtypes")
+            final AtomicLongFieldUpdater<ObserveOnSubscriber> counter = COUNTER_UPDATER;
```

Comment:
Curious, what benefit does this give to assign this reference? Why use `counter` instead of `COUNTER_UPDATER` directly?

Same for the other assignments here.